app/ide-desktop/lib/dashboard/vite.config.ts

```diff
@@ -5,48 +5,49 @@ import vitePluginYaml from '@modyfi/vite-plugin-yaml'
 import vitePluginReact from '@vitejs/plugin-react'
 import * as vite from 'vite'
 
-// =================
-// === Constants ===
-// =================
+import * as common from 'enso-common'
 
-const SERVER_PORT = 8080
+import * as env from './env'
+import * as globals from './globals'
 
 // =====================
 // === Configuration ===
 // =====================
 
-/* eslint-disable @typescript-eslint/naming-convention */
+await env.loadEnv()
 
-export default vite.defineConfig({
-  server: { port: SERVER_PORT },
-  plugins: [
-    vitePluginReact({
-      include: '**/*.tsx',
-      babel: {
-        plugins: ['@babel/plugin-syntax-import-assertions'],
+/* eslint-disable @typescript-eslint/naming-convention */
+export default vite.defineConfig(viteEnv => {
+  const { mode } = viteEnv
+  const serverPort = 8080
+  const devMode = mode === 'development'
+  return {
+    server: { port: serverPort, headers: Object.fromEntries(common.COOP_COEP_CORP_HEADERS) },
+    plugins: [
+      vitePluginReact({
+        include: '**/*.tsx',
+        babel: { plugins: ['@babel/plugin-syntax-import-assertions'] },
+      }),
+      vitePluginYaml(),
+    ],
+    resolve: {
+      alias: {
+        '#': url.fileURLToPath(new URL('./src', import.meta.url)),
       },
-    }),
-    vitePluginYaml(),
-  ],
-  resolve: {
-    alias: {
-      '#': url.fileURLToPath(new URL('./src', import.meta.url)),
     },
-  },
-  build: {
-    rollupOptions: {
-      input: {
-        main: url.fileURLToPath(new URL('./index.html', import.meta.url)),
-        '404': url.fileURLToPath(new URL('./404.html', import.meta.url)),
+    build: {
+      rollupOptions: {
+        input: {
+          main: url.fileURLToPath(new URL('./index.html', import.meta.url)),
+          '404': url.fileURLToPath(new URL('./404.html', import.meta.url)),
+        },
       },
     },
-  },
-  define: {
-    IS_VITE: JSON.stringify(true),
-    REDIRECT_OVERRIDE: JSON.stringify(`http://localhost:${SERVER_PORT}`),
-    CLOUD_ENV:
-      process.env.ENSO_CLOUD_ENV != null ? JSON.stringify(process.env.ENSO_CLOUD_ENV) : 'undefined',
-    // Single hardcoded usage of `global` in by aws-amplify.
-    'global.TYPED_ARRAY_SUPPORT': JSON.stringify(true),
-  },
+    define: {
+      IS_VITE: JSON.stringify(true),
+      // The sole hardcoded usage of `global` in by aws-amplify.
```

Comment:
`in by aws-amplify` seems wrong but I'm not sure what the comment should actually say 😅

Reply:
i think it's supposed to be just `in`